### PR TITLE
FE-1420 remove InputDecoratorBridge from Select

### PR DESCRIPTION
**Pull request by edleeks87**
# Description
Removes class names and implements multi-selected pills with styled-components
Updates component to allow removal of InputDecoratorBridge and replace with Textbox

### CODE REVIEW

File: src/__experimental__/components/select/select.stories.js [hunk 1]
```diff
@@ -13,34 +13,91 @@ const multiSelectStore = new Store({
...
+    placeholder: text('placeholder'),
+    readOnly: boolean('readOnly', false),
+    size: select('size', OptionsHelper.sizesRestricted),
+    warningMessage: text('warningMessage'),
+    filterable,
+    typeAhead
+  };
+};
+
+const selectOptions = [
```
Comment (izakrzewska) [inline src/__experimental__/components/select/select.stories.js]: isn't it possible to map through all the values the code is not repeated?

File: src/__experimental__/components/select/select.component.js [hunk 2]
```diff
@@ -188,88 +170,163 @@ class Select extends React.Component {
...
 }
+
+const optionShape = PropTypes.shape({
+  value: PropTypes.string.isRequired,
+  text: PropTypes.string.isRequired
+});
+
+Select.propTypes = {
+  children: PropTypes.node,
+  className: PropTypes.string,
```
Comment (izakrzewska) [inline src/__experimental__/components/select/select.component.js]: // legacy classname

File: src/__experimental__/components/select/select.style.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+import baseTheme from '../../../style/themes/base';
+import { isClassic } from '../../../utils/helpers/style-helper';
+
+const StyledSelectPillContainer = styled.div`
```
Comment (izakrzewska) [inline src/__experimental__/components/select/select.style.js]: couldn't it be styled(Pill)?

File: src/__experimental__/components/select/select-list.component.js [hunk 4]
```diff
@@ -111,4 +141,17 @@ class SelectList extends React.Component {
   }
 }
 
+SelectList.propTypes = {
```
Comment (dawidzarzycki) [inline src/__experimental__/components/select/select-list.component.js]: Would you like to add descriptions to the props like we have in the others components?

File: src/__experimental__/components/select/select.component.js [hunk 5]
```diff
@@ -188,88 +170,163 @@ class Select extends React.Component {
...
     );
   }
 }
+
+const optionShape = PropTypes.shape({
+  value: PropTypes.string.isRequired,
+  text: PropTypes.string.isRequired
+});
+
+Select.propTypes = {
```
Comment (dawidzarzycki) [inline src/__experimental__/components/select/select.component.js]: Would you like to add description to the props?

File: src/__experimental__/components/select/select.style.js (hunk 3 above)
Comment (edleeks87) [reply src/__experimental__/components/select/select.style.js]: The updated Pill component has not been merged yet so I had to go with this for now

File: src/__experimental__/components/select/select.component.js (hunk 2 above)
Comment (edleeks87) [reply src/__experimental__/components/select/select.component.js]: Yeah good spot that can be removed

File: src/__experimental__/components/select/select.stories.js (hunk 1 above)
Comment (kyanmichael) [reply src/__experimental__/components/select/select.stories.js]: Yes I believe so. I've refactored this in 54ecf5e :thumbsup:

File: src/__experimental__/components/select/select.component.js (hunk 2 above)
Comment (kyanmichael) [reply src/__experimental__/components/select/select.component.js]: Removed in 2444be9 :thumbsup:

File: src/__experimental__/components/select/select.component.js (hunk 5 above)
Comment (kyanmichael) [reply src/__experimental__/components/select/select.component.js]: Added in 2894e25 :thumbsup:

File: src/__experimental__/components/select/select-list.component.js (hunk 4 above)
Comment (kyanmichael) [reply src/__experimental__/components/select/select-list.component.js]: Added in 1418cb4 :thumbsup:

File: src/__experimental__/components/select/select.style.js (hunk 3 above)
Comment (edleeks87) [reply src/__experimental__/components/select/select.style.js]: I have updated it to use the new `StyledPill` https://github.com/Sage/carbon/pull/2163/commits/f3a8ce2624caf0670a23ae6ee8dca5c0dddfeed3

**Comment by harpalsingh** [comment]
Margin issue on single select in classic design, margin offset on the right div needs fixing.

Screenshot 2019-07-11 at 09 44 02

Is the hover color for Theme small single select correct?

Screenshot 2019-07-11 at 10 00 32

Currently has #E6E9E7

Focus on pill state in Theme Small? in Multi select

Screenshot 2019-07-11 at 10 02 18

**Comment by harpalsingh** [comment]
A minor thing when you load the component, the knobs have "small" as preselected, yet the size that is actually shown is medium, you then need to choose medium/large and then small again to get the real small size visual.

**Pull request merged**